### PR TITLE
Use custom JsonConverters for serializing/deserializing TagHelperResolutionResult

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #1094 

@NTaylorMullen @rynowak

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.Remote.Razor/Microsoft.CodeAnalysis.Remote.Razor.csproj [hunk 1]
```diff
@@ -11,8 +11,12 @@
     <PackageTargetFallback>$(PackageTargetFallback);portable-net45+win8+wp8+wpa81;</PackageTargetFallback>
   </PropertyGroup>
   <ItemGroup>
+    <Compile Include="..\Microsoft.VisualStudio.LanguageServices.Razor\RazorDiagnosticJsonConverter.cs" />
```
Comment (ajaybhargavb) [inline src/Microsoft.CodeAnalysis.Remote.Razor/Microsoft.CodeAnalysis.Remote.Razor.csproj]: This was needed so we don't have to introduce any new dependencies.

File: src/Microsoft.VisualStudio.LanguageServices.Razor/Resources.resx [hunk 2]
```diff
@@ -117,9 +117,6 @@
   <resheader name="writer">
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
-  <data name="RazorDiagnosticJsonConverter_UnsupportedRazorDiagnosticType" xml:space="preserve">
```
Comment (ajaybhargavb) [inline src/Microsoft.VisualStudio.LanguageServices.Razor/Resources.resx]: Spoke to @rynowak. This was a corner case that we don't really care about.

File: src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperResolver.cs [hunk 3]
```diff
@@ -83,6 +87,15 @@ internal class DefaultTagHelperResolver : ITagHelperResolver
             }
         }
 
+        private TagHelperResolutionResult GetTagHelperResolutionResult(JObject jsonObject)
+        {
+            var serializer = new JsonSerializer();
+            serializer.Converters.Add(TagHelperDescriptorJsonConverter.Instance);
+            serializer.Converters.Add(RazorDiagnosticJsonConverter.Instance);
+
+            return serializer.Deserialize<TagHelperResolutionResult>(jsonObject.CreateReader());
```
Comment (NTaylorMullen) [inline src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperResolver.cs]: Just double checking. No missing disposes happening here right?

Comment (ajaybhargavb) [inline src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperResolver.cs]: Good point 👍. Don't want to rely on the serializer to handle dispose

**Pull request merged**